langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithJsonSchemaIT.java

```diff
@@ -71,26 +73,325 @@ void should_extract_pojo_with_primitives() {
+
+            // then
+            assertThat(characteristics).hasSize(4);
+
+            // Use assertThat with filters to ensure each weather characteristic is in the list
+            ArrayList<WeatherExtractor.WeatherCharacteristic> expectedCharacteristics = new ArrayList<>();
+            expectedCharacteristics.add(WeatherExtractor.WeatherCharacteristic.SUNNY);
+            expectedCharacteristics.add(WeatherExtractor.WeatherCharacteristic.RAINY);
+            expectedCharacteristics.add(WeatherExtractor.WeatherCharacteristic.CLOUDY);
+            expectedCharacteristics.add(WeatherExtractor.WeatherCharacteristic.WINDY);
+            Assertions.assertTrue(characteristics.containsAll(expectedCharacteristics));
+
+            verify(model).chat(ChatRequest.builder()
+                .messages(singletonList(userMessage(text)))
+                .responseFormat(ResponseFormat.builder()
+                    .type(JSON)
+                    .jsonSchema(JsonSchema.builder()
+                        .name("Set_of_WeatherCharacteristic")
+                        .rootElement(JsonObjectSchema.builder()
+                            .addProperty("items", JsonArraySchema.builder()
+                                .items(JsonEnumSchema.builder()
+                                    .enumValues("SUNNY", "RAINY", "CLOUDY", "WINDY")
                                     .build())
+                                .build())
+                            .required("items")
                             .build())
-                    .build());
+                        .build())
+                    .build())
+                .build());
             verify(model).supportedCapabilities();
         }
     }
 
 
+    interface StatusExtractor {
+
+        enum MaritalStatus {
+
+            SINGLE, MARRIED
+        }
+
+        MaritalStatus extractMaritalStatusFrom(String text);
+    }
+
+    @Test
+    void should_extract_enum() {
+
+        for (ChatLanguageModel model : models()) {
+
+            // given
+            model = spy(model);
+
+            StatusExtractor personExtractor = AiServices.create(StatusExtractor.class, model);
+
+            String text = "Klaus is 37 years old, 1.78m height and single.";
+
+            // when
+            StatusExtractor.MaritalStatus status = personExtractor.extractMaritalStatusFrom(text);
+
+            // then
+            assertThat(status).isEqualTo(StatusExtractor.MaritalStatus.SINGLE);
+
+            verify(model).chat(ChatRequest.builder()
+                .messages(singletonList(userMessage(text)))
+                .responseFormat(ResponseFormat.builder()
+                    .type(JSON)
+                    .jsonSchema(JsonSchema.builder()
+                        .name("MaritalStatus")
+                        .rootElement(JsonObjectSchema.builder()
+                            .addEnumProperty("MaritalStatus", List.of("SINGLE", "MARRIED"))
```

Comment:
```suggestion
                            .addEnumProperty("value", List.of("SINGLE", "MARRIED"))
```
Let's use "value" instead, should be generic enough